src/commands/createWebApp/setDefaultRgAndPlanName.ts

```diff
@@ -10,54 +10,64 @@ import { MessageItem } from "vscode";
+import { getResourceGroupFromId } from '../../utils/azure';
 import { nonNullProp } from "../../utils/nonNull";
 import { getWorkspaceSetting, updateGlobalSetting } from "../../vsCodeConfig/settings";
+import { AzConfig, AzConfigProperties, readAzConfig } from "./readAzConfig";
 
 const maxNumberOfSites: number = 3;
 
 export async function setDefaultRgAndPlanName(wizardContext: IAppServiceWizardContext, siteNameStep: SiteNameStep): Promise<void> {
+    const config: AzConfig = await readAzConfig(wizardContext, AzConfigProperties.group);
+
     // this should always be set when in the basic creation scenario
     const location: Location = nonNullProp(wizardContext, 'location');
     const defaultName: string = `appsvc_${wizardContext.newSiteOS}_${location.name}`;
+    const defaultGroupName: string = config.group ? config.group : defaultName;
+    const defaultPlanName: string = defaultName;
 
-    const asp: AppServicePlan | null = await getAppServicePlan(wizardContext, defaultName, defaultName);
+    const asp: AppServicePlan | null = await getAppServicePlan(wizardContext, defaultGroupName, defaultPlanName);
     if (asp && checkPlanForPerformanceDrop(asp)) {
         // Subscriptions can only have 1 free tier Linux plan so show a warning if there are too many apps on the plan
         if (wizardContext.newSiteOS === WebsiteOS.linux) {
             await promptPerformanceWarning(wizardContext, asp);
-            wizardContext.newResourceGroupName = defaultName;
-            wizardContext.newPlanName = defaultName;
+            wizardContext.newResourceGroupName = defaultGroupName;
+            wizardContext.newPlanName = defaultPlanName;
         } else {
             // Subscriptions can have 10 free tier Windows plans so just create a new one with a suffixed name
             // If there are 10 plans, it'll throw an error that directs them to advanced create
 
             const client: WebSiteManagementClient = createAzureClient(wizardContext, WebSiteManagementClient);
             const allAppServicePlans: AppServicePlan[] = await client.appServicePlans.list();
-            const defaultPlans: AppServicePlan[] = allAppServicePlans.filter(plan => { return plan.name && plan.name.includes(defaultName); });
+            const defaultPlans: AppServicePlan[] = allAppServicePlans.filter(plan => { return plan.name && plan.name.includes(defaultPlanName); });
 
             // when using appServicePlans.list, the numOfSites are all set to 0 so individually get each plan and look for one with less than 3 sites
             for (const plan of defaultPlans) {
                 if (plan.name) {
-                    const fullPlanData: AppServicePlan | null = await getAppServicePlan(wizardContext, plan.name, plan.name);
+                    const groupName: string = getResourceGroupFromId(nonNullProp(plan, 'id'));
+                    const fullPlanData: AppServicePlan | null = await getAppServicePlan(wizardContext, groupName, plan.name);
```

Comment:
I feel like this should check `if (groupName.includes(defaultGroupName))` - otherwise you might run into weird behavior if you have a plan named the same thing in different resource groups. Here are some repro steps:
1. Remove az cli defaults
1. Create a web app
1. Add az cli defaults for group name 'myGroup'
1. Create 4 web apps

Expected:
The 4th web app will create a new plan in group 'myGroup'

Actual:
The 4th web app creates a new plan in 'appsvc_windows_eastus' (or whatever) from step 1